[json] Add missing "status" string for VideoLibrary.GetTVShowDetails

sualfred opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->
Setting the string is possible since early 2016 but it was forgotten to add it as readable value.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
See https://forum.kodi.tv/showthread.php?tid=348889&pid=2898553#pid2898553

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [] All new and existing tests passed

sualfred commented:
@DaveTBlake 

Like so?

rmrector commented:
Thumbs up on the addition.

Any thoughts on a short description that could be added (to this and the existing param on "SetTVShowDetails")? `status` is quite a generic identifier. "runtime" right above has an example.

DaveTBlake commented on xbmc/interfaces/json-rpc/schema/version.txt:
```diff
@@ -1 +1 @@
-JSONRPC_VERSION 10.5.2
+JSONRPC_VERSION 10.5.3
```
Adding a property at the end is a non-breaking change to the API, hence bump the minor (middle) number 10.6.0
See https://github.com/xbmc/xbmc/blob/88fffa1392bd59914a4050d0a52ab6799c11a54b/xbmc/interfaces/json-rpc/schema/methods.json#L33

sualfred replied:
Changed.

DaveTBlake commented on xbmc/interfaces/json-rpc/schema/types.json:
```diff
@@ -112,7 +112,7 @@
       "order": { "type": "string", "default": "ascending", "enum": [ "ascending", "descending" ] },
       "ignorearticle": { "type": "boolean", "default": false },
-      "useartistsortname": { "type": "boolean", "default": false }	  
+      "useartistsortname": { "type": "boolean", "default": false }
```
A whitespace change to an area that you aren't otherwise touching. This makes git blame hard to read etc., so can you drop the white space clean up that your editor has done here and a few other places

sualfred replied:
Feels wrong to readd a valid cleanup, but I agree that it makes it harder to read. Reverted.

DaveTBlake reviewed (changes requested):
Thanks for this @sualfred, almost right first time :)

sualfred committed:
[json] Add missing "status" string for VideoLibrary.GetTVShowDetails

Setting the string is possible since early 2016 but it was forgotten to add it as readable value.

sualfred commented:
> Thumbs up on the addition.
> 
> Any thoughts on a short description that could be added (to this and the existing param on "SetTVShowDetails")? `status` is quite a generic identifier. "runtime" right above has an example.

Done. I've added it to methods as well.

sualfred commented:
@DaveTBlake 
Can we backport this to Leia, too?

DaveTBlake commented:
Regarding whitespace clean up, if you want to raise a clean-up PR to do this then please go ahead :)

Thanks for the desc additions, more of the API could benefit from those. Just need to get some TV show data so I can test it.

DaveTBlake commented:
>Can we backport this to Leia, too?

Is something missing for 3 years a fix? I lean towards thinking it is and as such would say yes. However, there are other API changes in v19 master that have not been backported and I'm not sure how to handle that deviation under schema version numbering. Since it has been waiting for 3 years it doesn't seem urgent either. But I am open to pursasive arguements

sualfred commented:
> > Can we backport this to Leia, too?
> 
> Is something missing for 3 years a fix? I lean towards thinking it is and as such would say yes. However, there are other API changes in v19 master that have not been backported and I'm not sure how to handle that deviation under schema version numbering. Since it has been waiting for 3 years it doesn't seem urgent either. But I am open to pursasive arguements

I just would require it for my metadata editor script that is working for Leia <. That's why I even noticed this missing field.

DaveTBlake commented:
Understood, but because we did not bump major API version when we forked to v19 and have changed the API without backporting, I really don't see how to backport this. I will give it some thought

sualfred commented:
Thanks, would be great if it's somehow possible. If not I have to live with it.

DaveTBlake commented:
jenkins build this please

DaveTBlake commented:
Build failure is UWP/Py3 related, not this PR
